music_assistant/server/providers/plex/__init__.py

```diff
@@ -0,0 +1,467 @@
+        )
+        if plex_album.year:
+            album.year = plex_album.year
+        if thumb := plex_album.firstAttr("thumb", "parentThumb", "grandparentThumb"):
+            album.metadata.images = [MediaItemImage(ImageType.THUMB, thumb, self.instance_id)]
+        if plex_album.summary:
+            album.metadata.description = plex_album.summary
+
+        album.artist = await self._get_data_cached(plex_album.parentKey, Artist)
+
+        album.add_provider_mapping(
+            ProviderMapping(
+                item_id=str(album_id),
+                provider_domain=self.domain,
+                provider_instance=self.instance_id,
+                url=plex_album.getWebURL(),
+            )
+        )
+        return album
+
+    async def _parse_artist(self, plex_artist: PlexArtist) -> Artist:
+        """Parse a Plex Artist response to Artist model object."""
+        artist_id = plex_artist.key
+        if not artist_id:
+            raise InvalidDataError("Artist does not have a valid ID")
+        artist = Artist(item_id=artist_id, name=plex_artist.title, provider=self.domain)
+        if plex_artist.summary:
+            artist.metadata.description = plex_artist.summary
+        if thumb := plex_artist.firstAttr("thumb", "parentThumb", "grandparentThumb"):
+            artist.metadata.images = [MediaItemImage(ImageType.THUMB, thumb, self.instance_id)]
+        artist.add_provider_mapping(
+            ProviderMapping(
+                item_id=str(artist_id),
+                provider_domain=self.domain,
+                provider_instance=self.instance_id,
+                url=plex_artist.getWebURL(),
+            )
+        )
+        return artist
+
+    async def _parse_playlist(self, plex_playlist: PlexPlaylist) -> Playlist:
+        """Parse a Plex Playlist response to a Playlist object."""
+        playlist = Playlist(
+            item_id=plex_playlist.key, provider=self.domain, name=plex_playlist.title
+        )
+        if plex_playlist.summary:
+            playlist.metadata.description = plex_playlist.summary
+        if thumb := plex_playlist.firstAttr("thumb", "parentThumb", "grandparentThumb"):
+            playlist.metadata.images = [MediaItemImage(ImageType.THUMB, thumb, self.instance_id)]
+        playlist.is_editable = True
+        playlist.add_provider_mapping(
+            ProviderMapping(
+                item_id=plex_playlist.key,
+                provider_domain=self.domain,
+                provider_instance=self.instance_id,
+                url=plex_playlist.getWebURL(),
+            )
+        )
+        return playlist
+
+    async def _parse_track(self, plex_track: PlexTrack) -> Track:
+        """Parse a Plex Track response to a Track model object."""
+        track = Track(item_id=plex_track.key, provider=self.domain, name=plex_track.title)
+
+        if plex_track.grandparentKey:
+            track.artist = await self._get_data_cached(plex_track.grandparentKey, Artist)
```

Comment:
same comment as for the album - you may just return an ItemMapping here if you have at least some data